src/normalizer/japanese.rs

```diff
@@ -0,0 +1,194 @@
+use std::borrow::Cow;
+
+use super::{Normalizer, NormalizerOption};
+use crate::detection::{Language, Script};
+use crate::Token;
+use wana_kana::is_hiragana::*;
+use wana_kana::to_hiragana::to_hiragana_with_opt;
+use wana_kana::Options;
+
+/// Japanese specialized [`Normalizer`].
+///
+/// This Normalizer uses [`wana_kana`] internally to convert Katakana and Kanji to Hiragana
+pub struct JapaneseNormalizer;
+
+impl Normalizer for JapaneseNormalizer {
+    fn normalize<'o>(
+        &self,
+        mut token: Token<'o>,
+        options: NormalizerOption,
+    ) -> Box<dyn Iterator<Item = Token<'o>> + 'o> {
+        if is_hiragana(token.lemma()) {
+            // No need to convert
+
+            if options.create_char_map && token.char_map.is_none() {
+                let mut char_map = Vec::new();
+                for c in token.lemma().chars() {
+                    char_map.push((c.len_utf8() as u8, c.len_utf8() as u8));
+                }
+                token.char_map = Some(char_map);
+            }
+        } else {
```

Comment:
Having an `Identity` char map is the same that not having it, so don't bother creating it.
```suggestion
        if !is_hiragana(token.lemma()) {
```

Reply:
Oh, I originally thought we must create one when `create_char_map` is true 😅  Seems this is not the case, so the implementation can be much simpler. Thanks for the suggestion!

Reply:
Addressed in 49bc172